src/Apps/Order/Routes/Shipping2/Shipping2.jest.tsx

```diff
@@ -829,29 +845,47 @@ describe("Shipping", () => {
 
             // Clicking "Back to Edit" allows users to edit the address form
             // and requires clicking "Save and Continue" to proceed.
-            userEvent.click(screen.getByText("Back to Edit"))
-            await userEvent.click(screen.getByText("Save and Continue"))
-            mockCommitMutation.mockResolvedValueOnce(
-              settingOrderShipmentSuccess
-            )
-            expect(mockCommitMutation).toHaveBeenCalledTimes(1)
-
-            let mutationArg = mockCommitMutation.mock.calls[0][0]
-            expect(mutationArg.mutation.default.operation.name).toEqual(
-              "SetShippingMutation"
+            await userEvent.click(screen.getByText("Back to Edit"))
+            await userEvent.paste(
+              screen.getByPlaceholderText("City"),
+              ": the big apple"
             )
-            expect(mutationArg.variables).toEqual({
+            await userEvent.click(screen.getByText("Save and Continue"))
+            mockCommitMutation
+              .mockResolvedValueOnce(settingOrderShipmentSuccess)
+              .mockResolvedValueOnce(updateAddressSuccess)
+            await waitFor(() => {
+              expect(mockCommitMutation).toHaveBeenCalledTimes(2)
+            })
+            let fulfillmentMutationArg = mockCommitMutation.mock.calls[0][0]
+            expect(
+              fulfillmentMutationArg.mutation.default.operation.name
+            ).toEqual("SetShippingMutation")
+            expect(fulfillmentMutationArg.variables).toEqual({
               input: {
                 id: "1234",
                 fulfillmentType: "SHIP",
                 addressVerifiedBy: "USER",
                 phoneNumber: validAddress.phoneNumber,
                 shipping: {
-                  ...validAddress,
+                  ...initialAddress,
+                  city: "New York: the big apple",
                   phoneNumber: "",
                 },
               },
             })
+            const addressMutationArg = mockCommitMutation.mock.calls[1][0][1]
+            expect(addressMutationArg.mutation.default.operation.name).toEqual(
```

Comment:
strange: The nesting or args changes between 1st and 2nd call. I don't know why but i am pretty sure i've seen this before in the same file.